modules/ROOT/pages/http/http-listener-ref.adoc

```diff
@@ -1,182 +1,296 @@
+two way authentication is desired. See the xref:4.1@mule-runtime/tls-configuration.adoc[TLS Configuration] guide for details.
 
-== Setting the Path
+[source,xml,linenums]
+----
+<http:listener-connection host="0.0.0.0" port="8081" protocol="HTTPS">
+  <tls:context>
+    <tls:trust-store path="keystore.jks" password="MyP455W0rD"/>
+  </tls:context>
+</http:listener-connection>
+----
 
-If a request matches the path of multiple Listeners, the Listener having the most specific path has precedence, regardless of the order in which the operations are defined. Only one connector handles each request.
-For example, given the following Listener configurations, the second one receives the request when you use this URL: `account/mulesoft/main-contact`
+[TIP]
+===
+To deploy a project to CloudHub, setting the host to *0.0.0.0* is required since it sets up the connection to listen on all interfaces of the machine.
 
-* Path = `account/(...)`
-* Path = `account/(...)/main-contact`
+When you deploy locally for testing purposes, setting the host to *localhost* is better since it means only requests generated inside the server (your local machine) will be received: the app is not vulnerable to external threats.
+===
 
-=== Wildcard in Path
+Advanced users can also define whether the connections will be persistent and if not, the timeout they will have.
+
+== HTTP Listener
+
+The HTTP listener is configured as a source within flows and must reference the configuration it will use and the path where it will listen for requests.
+
+[source,xml,linenums]
+----
+<flow name="http-triggered-flow">
+  <http:listener path="/" config-ref="HTTP_Listener_config"/>
+  <set-payload value="Help! I've been triggered!"/>
+</flow>
+----
 
-In the Path setting, you can use the wildcard asterisk (*) as follows:
+When an HTTP request is matched to a listener, its request data is used to trigger the flow. The body is set as the payload while all other metadata from the request line to the headers are shared as attributes.
+When the triggered flow is done, the HTTP listener uses it's result to generate a proper HTTP response.
 
-* To listen for incoming requests to a path within the specified base path. Use a wildcard at the end of a path.
-* To listen for all requests for a resource in a URI. Use a wildcard in the middle of a path.
+=== About Paths
 
-You can specify a partial path that ends with an asterisk, for example: `/somepath/*`. Any request succeeds that has a URL consisting of the prefix to the asterisk (`/somepath`).
+The path of an HTTP listener can be static, which requires exact matches, or feature placeholders. Placeholders can be wildcards (`*`), which will match against anything it's compared to, or parameters (`{param}`), which will not only match against anything but also capture those values on an URI parameters map.
```

Comment:
`path params` is confusing since the term `URI params` is already established and similar. I don't see how to make this clearer other than with examples as the ones below.

Reply:
Perhaps just saying the path can be static or parameterized?